### PR TITLE
[4.3][csp] Reduce inline onclick usage [List views]

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- Remove the inline onclick scripts

### Testing Instructions

wip

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on e56e30954faa480902ab0abc79b324b9da9b7bed<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39730">issues.joomla.org/tracker/joomla-cms/39730</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 79285eb145200ea3f6ffd903136569a8b46a78ef<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39730">issues.joomla.org/tracker/joomla-cms/39730</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39730">issues.joomla.org/tracker/joomla-cms/39730</a>.</sub>

**Comment by obuisard** [comment]
Thank you Dimitris @dgrammatiko !

File: libraries/src/HTML/Helpers/Grid.php [hunk 1]
```diff
@@ -56,13 +56,13 @@ public static function sort($title, $order, $direction = 'asc', $selected = '',
...
-        $html = '<a href="#" onclick="Joomla.tableOrdering(\'' . $order . '\',\'' . $direction . '\',\'' . $task . '\'' . $form . ');return false;"'
-            . ' class="hasPopover" title="' . htmlspecialchars(Text::_($tip ?: $title)) . '"'
-            . ' data-bs-content="' . htmlspecialchars(Text::_('JGLOBAL_CLICK_TO_SORT_THIS_COLUMN')) . '" data-bs-placement="top">';
+        $html = '<a href="#" class="hasPopover" title="' . htmlspecialchars(Text::_($tip ?: $title)) . '"'
+            . ' data-ordering-order="' . $order . '"'
+            . ' data-ordering-direction="' . $direction . '"'
+            . ' data-ordering-task="' . $task . '"'
+            . ' data-ordering-form="' . (isset($form) ? $form : '') . '"'
+            . ' data-bs-content="' . htmlspecialchars(Text::_('JGLOBAL_CLICK_TO_SORT_THIS_COLUMN')) . '"'
+            . ' data-bs-placement="top">';
```
Comment (Quy) [inline libraries/src/HTML/Helpers/Grid.php:66]: With the upmerge of #39643, sorting is now broken on the frontend. Can you please look into this? Thank you.

**Pull request merged**